examples/testdata/testframework/impl/framework.go

```diff
@@ -18,27 +18,54 @@ type TestDevbox struct {
 	devboxJsonPath string
 }
 
-func (td *TestDevbox) Info(pkg string, markdown bool) (string, error) {
-	cmd := boxcli.InfoCmd()
-	output, err := runCmd(cmd, []string{pkg}, false)
+func (td *TestDevbox) SetDevboxJson(path string) error {
+	td.devboxJsonPath = path
+	return nil
+}
+
+func (td *TestDevbox) GetDevboxJson() (*impl.Config, error) {
+	file, err := os.ReadFile(td.devboxJsonPath)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	data := &impl.Config{}
+	err = json.Unmarshal(file, &data)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	return data, nil
+}
+
+func (td *TestDevbox) Add(pkgs ...string) (string, error) {
+	cmd := boxcli.AddCmd()
+	output, err := runCmd(cmd, pkgs)
 	if err != nil {
 		return "", errors.WithStack(err)
 	}
 	return string(output), nil
 }
 
-func (td *TestDevbox) Version() (string, error) {
-	cmd := boxcli.VersionCmd()
-	output, err := runCmd(cmd, nil, false)
+func (td *TestDevbox) Generate(subcommand string) (string, error) {
+	cmd := boxcli.GenerateCmd()
+	output, err := runCmd(cmd, []string{subcommand})
 	if err != nil {
 		return "", errors.WithStack(err)
 	}
 	return string(output), nil
 }
 
-func (td *TestDevbox) Add(pkgs ...string) (string, error) {
-	cmd := boxcli.AddCmd()
-	output, err := runCmd(cmd, pkgs, false)
+func (td *TestDevbox) Info(pkg string, markdown bool) (string, error) {
+	cmd := boxcli.InfoCmd()
+	output, err := runCmd(cmd, []string{pkg})
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	return string(output), nil
+}
+
+func (td *TestDevbox) Init() (string, error) {
```

Comment:
this is the only function I added. The rest of the changes in this file are due to re-organizing the order of funcs to the same order that they are listed in interface.go